src/app/group-sort.pipe.ts

```diff
@@ -87,14 +106,38 @@ export class GroupSort implements PipeTransform {
     return groupedArr;
   }
 
-  private lexicographicComp(first: LinkData, second: LinkData): number {
-    if (first.href < second.href) {
+  /**
+   * Comparison function for arbitrary objects that compares lexicographically
+   * based on the attributes of the object corresponding to key.
+   * @param { any } first
+   * @param { any } second
+   * @param { string } key The key denoting which attribute should be used for
+   * the comparison.
+   * @return { number }
+   */
+  private lexicographicComp(first: any, second: any, key: string): number {
```

Comment:
This is designed to only be used with objects that have specific attribute. For example since we know that all LinkData has an href, we use this with a key of 'href' on LinkData object. If you use this with objects that don't have the provided key then the behavior is undefined.

Reply:
Right, and it's not preferable to have undefined behavior. I would create a comparator for each type instead of making it generic.

Reply:
okay